### PR TITLE
remove reference to dynamo0.1 api in the documentation (part of #282)

### LINKED ISSUE #282: Remove the dynamo 0.1 API
Dynamo 0.1 has been superseded by 0.3 and has not been developed for several years.

### PR DESCRIPTION
**Pull request by rupertford**

### CODE REVIEW

**Comment by rupertford** [comment]
Turns out there is very little reference to dynamo0.1 in the documentation so this is a pleasantly small PR.

**Comment by TeranIvy** [comment]
Happy to take this, as well as #1599 when it's ready.

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Thanks for taking this on. All looks straightforward. I ran the tests with compilation to be on the safe side and I also built the User and the Developer guide. All was well.

I have a couple of suggestions inline. I also grepped for `0.1` and `0p1` in the documentation and found one occurence of `dynamo0.1` in the Develope guide and a few occurences of `gocean0.1` API. I am aware that the Gocean 0.1 API is being removed in another PR, however I am listing what I found here for reference.

Mentions of Dynamo 0.1: `developer_guide/APIs.rst:53:.. currently supports the following APIs; dynamo versions 0.1 and 0.3`

Mentions of GOcean 0.1:
```
developer_guide/APIs.rst:54:.. and gocean versions 0.1 and 1.0.
thebigpicture.mm:128:<node COLOR="#000000" CREATED="0" MODIFIED="0" TEXT="gocean0.1">
user_guide/configuration.rst:145:                        "gocean0.1", "gocean1.0" and "nemo"). If there is no
user_guide/psyclone_command.rst:83:                          'gocean0.1', 'gocean1.0', 'nemo'],
```

File: doc/user_guide/psy_layer.rst [hunk 1]
```diff
@@ -92,8 +92,8 @@ and the PSy layer code being generated by the PSy-generation object.
     from psyclone.parse.algorithm import parse
     from psyclone.psyGen import PSyFactory
     
-    # This example uses version 0.1 of the Dynamo API
-    api = "dynamo0.1"
+    # This example uses version 0.3 of the Dynamo API
```
[the remark this answers is not in the document]
Comment (rupertford) [reply doc/user_guide/psy_layer.rst:95]: Good idea.

File: doc/user_guide/configuration.rst [hunk 2]
```diff
@@ -141,7 +141,7 @@ Entry                   Description
 ======================= =======================================================
 DEFAULTAPI              The API that PSyclone assumes an Algorithm/Kernel
                         conforms to if no API is specified. Must be one of the
-                        APIs supported by PSyclone ("dynamo0.1", "dynamo0.3",
+                        APIs supported by PSyclone ("dynamo0.3",
```
[the remark this answers is not in the document]
Comment (rupertford) [reply doc/user_guide/configuration.rst:144]: Modified.

**Comment by rupertford** [comment]
Sorry for missing the reference to dynamo0.1 in the developer's guide, I was a bit sloppy with the grep. Should be OK now. Ready for next review from @TeranIvy subject to CI happiness.

**Comment by TeranIvy** [comment]
I'll update this after #1578 is on.

**Review by TeranIvy — APPROVED** [review]
All well now. There are still some references to `dynamo0.1` in the generated documentation, however this is due to the auto-generated documentation of some transformation classes and should be resolved by #1599.

Proceeding to merge if the CI is happy.

**Pull request merged**